### PR TITLE
Strict lint rule for token camel case usage

### PR DESCRIPTION
**Pull request by libertymayc**
Lint rule only allows for defined CSS attributes and merged states (e.g. activeDisabled) to be allowed in tokens

### CODE REVIEW

File: packages/theme/stories/foundations/size.stories.mdx [hunk 1]
```diff
@@ -12,8 +12,10 @@ All size values are built on the basis of 4px, `--salt-size-basis-unit`. We enco
 
 <DocGrid>
   <SpacingBlock spacingVar="--salt-size-accent" />
+  <SpacingBlock spacingVar="--salt-size-adornment-gap" />
```
Comment (libertymayc) [inline packages/theme/stories/foundations/size.stories.mdx:15]: adornmentGap wasn't documented, so I don't feel the need to document the deprecation

Comment (joshwooding) [reply packages/theme/stories/foundations/size.stories.mdx:15]: I think the non-documentation was a bug. We should document it as deprecated :)

**Review by joshwooding — COMMENTED** [review]
Where did this change request come from? Was it noticed elsewhere? I thought that the style lint setup should catch these sort of issues?

**Review by feliciackh — APPROVED** [review]

**Comment by libertymayc** [comment]
> Where did this change request come from? Was it noticed elsewhere? I thought that the style lint setup should catch these sort of issues?

it wasn't that strict; but I've changed this PR to make it be

**Comment by libertymayc** [comment]
@joshwooding 
Added lint rule, which is fine for core - but if you look at the failing list I think we may not be able to do it like this, depending on how strict we want to go...

Tokens like `--calendar-day-outOfRange-text-color`, I think outOfRange is better than out-of-range, in this scenario (personal preference), anything that stands as one piece should be camel case - but things like `-adornmentGap` should be `-adornment-gap`, because it would make sense to have `-adornment-color`, `-adornment-size`... etc - which can only be determined by us if it makes sense or not, not a lint rule

Thoughts?

File: .changeset/famous-queens-dream.md [hunk 2]
```diff
@@ -0,0 +1,13 @@
+---
+"@salt-ds/core": minor
+"@salt-ds/lab": minor
+"@salt-ds/theme": minor
+---
+
+Lint rule to strictly only allow for CSS attributes and merged states (`activeDisabled`, `blurSelected`, `partialDisabled`, `selectedDisabled`) to be in camel case within tokens
+
+Deprecated `--size-salt-adornmentGap`; replaced with `--salt-size-adornment-gap`
```
Comment (origami-z) [inline .changeset/famous-queens-dream.md:9]: This just got [released yesterday](https://github.com/jpmorganchase/salt-ds/releases/tag/%40salt-ds%2Ftheme%401.3.0) and now we're asking the users to change again. 

I think it's time we take a pause and think about how to deal with theme token additional / change. 

@joshwooding @pseys

Comment (joshwooding) [reply .changeset/famous-queens-dream.md:9]: Totally agree, I've already raised this.

**Pull request closed without merging**